### PR TITLE
Fix/serviceworker cachebusting

### PR DESCRIPTION
**Pull request by vertexactual**
Majorly refactored the service worker implementation to be simpler and fresher.

Basically, this will always make the requests and when those fail, it'll use its cache.

### CODE REVIEW

File: service-worker.js [hunk 1]
```diff
@@ -1,66 +1,53 @@
...
+			fetch(request)
+			.catch((err) => {
+				console.error(err);
+				return caches.open(cacheKey).then((cache) => {
+					return cache.match(request);
+				});
+			})
+		)
+	}
 });
```
Comment (sarayourfriend) [inline service-worker.js:74]: Should we keep the cache clean up code? Or is it not really necessary?

File: service-worker.js [hunk 2]
```diff
@@ -1,66 +1,53 @@
...
-            }).map((cacheName) => {
-                return caches.delete(cacheName);
-            })
-        );
-      })
-    );
+	const { request } = event;
+	if (request.method === 'GET') {
+		event.respondWith(
+			fetch(request)
```
Comment (sarayourfriend) [inline service-worker.js:58]: I think `fetch` "consumes" the `request` doesn't it? Should we do `request.clone()`?

Also if this request is successful, should we update the cache with the new response?

**Review by sarayourfriend — COMMENTED** [review]
LGTM! Just a couple questions for my own understanding but otherwise this is great, I really appreciate your knowledge of these things 🙂

**Review by AetherUnbound — APPROVED** [review]
I can't comment much on the actual code, but I support the approach and the goal!

File: service-worker.js (hunk 1 above)
Comment (vertexactual) [reply service-worker.js:74]: Yup, good call. Added that back in.

File: service-worker.js (hunk 2 above)
Comment (vertexactual) [reply service-worker.js:58]: re: `request.clone()`: I saw varying boilerplate on whether or not to clone, and it didn't seem to make a difference in my testing in FF & Chrome, so I went with the simpler version. One to look out for, but afaik it's not necessary.

And also good call on updating the cache. I've added that in.

**Review by sarayourfriend — APPROVED** [review]
This is awesome! One last thing, should we update the `version` in the `config.yml` for this?

**Comment by vertexactual** [comment]
> This is awesome! One last thing, should we update the `version` in the `config.yml` for this?

I didn't update the version cause of not being sure if there's a protocol to actually do that. The version is used as the cache key now, so it's semi-important to make sure it does get changed with new changes. I'll push a commit with a new version.

**Review by AetherUnbound — APPROVED** [review]
Fantastic! Let's get this merged! :tada:

**Pull request merged**